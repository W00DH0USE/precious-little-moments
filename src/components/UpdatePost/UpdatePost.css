.modal-wrapper {
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.7);
  position: absolute;
  top: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;
}

.update-form {
  background-color: #fff;
  margin-top: 50px;
}

.button-continue,
.button-cancel {
  width: 100px;
  margin-left: 5px;
  margin-right: 5px;
  font-size: 20px;
}

.button-continue{
  background-color: greenyellow;
  border-color: greenyellow;
}

.button-cancel{
  background-color: red;
  border-color: red;
}
