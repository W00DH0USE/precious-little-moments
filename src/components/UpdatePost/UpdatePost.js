import React from 'react';
import { Form, Input2, Required, Textarea, Button } from "../Utils/Utils";
import './UpdatePost.css';

function UpdatePost(props) {

  const post = props.posts.filter(post => post.id === props.postId)
  return (
    <section>
      <div className='modal-wrapper updatePostModal'>
        <div className="modal-content">
          <Form className='update-form' onSubmit={props.handleUpdateSubmit}>
            <div className="moment-title">
              <h2 className="Form__title">Precious Little Moment: <span><em>update moment</em></span></h2>
              <hr />
              <label htmlFor="post_title">
                Moment Title <Required />
              </label>
              <Input2
                required
                type="text"
                name="post_title"
                id="post_title"
                defaultValue={post[0].post_title}
              ></Input2>
            </div>

            <div className="password">
              <label htmlFor="post_content">
                Content <Required />
              </label>
              <Textarea
                required
                name="post_content"
                id="post_content"
                defaultValue={post[0].post_content}
              ></Textarea>
            </div>
            <Button className="button demo-button button-continue" type="submit">Continue</Button>
            <Button className="button demo-button button-cancel" onClick={props.close}>Cancel</Button>
          </Form>
        </div>
      </div>
    </section>
  )
}

export default UpdatePost;