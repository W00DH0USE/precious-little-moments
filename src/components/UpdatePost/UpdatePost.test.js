import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UpdatePost from './UpdatePost';


describe(`Update Post Component`, () => {
  const post = [
    {
      post_title: 'Test Post Title 1',
      post_content: 'Test Post content 1',
      owner: 1
    },
    {
      post_title: 'Test Post Title 2',
      post_content: 'Test Post content 2',
      owner: 1
    },
    {
      post_title: 'Test Post Title 3',
      post_content: 'Test Post content 3',
      owner: 1
    },
  ]
  it('renders without crashing', () => {
    const wrapper = shallow(<UpdatePost posts={post} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});