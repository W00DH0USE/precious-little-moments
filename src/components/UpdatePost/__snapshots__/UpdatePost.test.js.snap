// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update Post Component renders without crashing 1`] = `
<section>
  <div
    className="modal-wrapper updatePostModal"
  >
    <div
      className="modal-content"
    >
      <Form
        className="update-form"
      >
        <div
          className="moment-title"
        >
          <h2
            className="Form__title"
          >
            Precious Little Moment: 
            <span>
              <em>
                update moment
              </em>
            </span>
          </h2>
          <hr />
          <label
            htmlFor="post_title"
          >
            Moment Title 
            <Required />
          </label>
          <Input2
            defaultValue="Test Post Title 1"
            id="post_title"
            name="post_title"
            required={true}
            type="text"
          />
        </div>
        <div
          className="password"
        >
          <label
            htmlFor="post_content"
          >
            Content 
            <Required />
          </label>
          <Textarea
            defaultValue="Test Post content 1"
            id="post_content"
            name="post_content"
            required={true}
          />
        </div>
        <Button
          className="button demo-button button-continue"
          type="submit"
        >
          Continue
        </Button>
        <Button
          className="button demo-button button-cancel"
        >
          Cancel
        </Button>
      </Form>
    </div>
  </div>
</section>
`;
